<a href="https://colab.research.google.com/github/jefferjone/food-sales-prediction/blob/main/Week_03_Machine_Learning_Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Week 03 Machine Learning: Project 2 - Part 1 (Core)

Submitted By: Jefferjone Long-gat

In [9]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [10]:
import pandas as pd


In [11]:
#First Choice
df1 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Coding Dojo/DATASET/Project/Project_Presentation/energy_efficiency_data.csv')
df1.head()


Relative_Compactness  Surface_Area  Wall_Area  Roof_Area  Overall_Height  \
0                  0.98         514.5      294.0     110.25             7.0   
1                  0.98         514.5      294.0     110.25             7.0   
2                  0.98         514.5      294.0     110.25             7.0   
3                  0.98         514.5      294.0     110.25             7.0   
4                  0.90         563.5      318.5     122.50             7.0   

   Orientation  Glazing_Area  Glazing_Area_Distribution  Heating_Load  \
0            2           0.0                          0         15.55   
1            3           0.0                          0         15.55   
2            4           0.0                          0         15.55   
3            5           0.0                          0         15.55   
4            2           0.0                          0         20.84   

   Cooling_Load  
0         21.33  
1         21.33  
2         21.33  
3         21.33  
4         28.28

In [12]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 10 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Relative_Compactness       768 non-null    float64
 1   Surface_Area               768 non-null    float64
 2   Wall_Area                  768 non-null    float64
 3   Roof_Area                  768 non-null    float64
 4   Overall_Height             768 non-null    float64
 5   Orientation                768 non-null    int64  
 6   Glazing_Area               768 non-null    float64
 7   Glazing_Area_Distribution  768 non-null    int64  
 8   Heating_Load               768 non-null    float64
 9   Cooling_Load               768 non-null    float64
dtypes: float64(8), int64(2)
memory usage: 60.1 KB


1) Source of data

https://www.kaggle.com/


2) Brief description of data

- The dataset contains parameters that can assess the cooling and heating loads of buildings.

3) What is the target?

- Target parameters can be the heating or cooling load

4) Is this a classification or regression problem?

- This is a regression problem

5) How many features?

- We can 9 features forr this problem

6) How many rows of data.

- There are 767 rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

- Some challenges for the dataset will be the right choice of model and the metrics to be used for  the model. Also, need to understand each parameter and how it will affect the model. Currently, no null values from the data. 

In [13]:
#Second Choice

df2 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Coding Dojo/DATASET/Project/Project_Presentation/water_potability.csv')
df2.head()


ph    Hardness        Solids  Chloramines     Sulfate  Conductivity  \
0       NaN  204.890455  20791.318981     7.300212  368.516441    564.308654   
1  3.716080  129.422921  18630.057858     6.635246         NaN    592.885359   
2  8.099124  224.236259  19909.541732     9.275884         NaN    418.606213   
3  8.316766  214.373394  22018.417441     8.059332  356.886136    363.266516   
4  9.092223  181.101509  17978.986339     6.546600  310.135738    398.410813   

   Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       10.379783        86.990970   2.963135           0  
1       15.180013        56.329076   4.500656           0  
2       16.868637        66.420093   3.055934           0  
3       18.436524       100.341674   4.628771           0  
4       11.558279        31.997993   4.075075           0

In [14]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3276 entries, 0 to 3275
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ph               2785 non-null   float64
 1   Hardness         3276 non-null   float64
 2   Solids           3276 non-null   float64
 3   Chloramines      3276 non-null   float64
 4   Sulfate          2495 non-null   float64
 5   Conductivity     3276 non-null   float64
 6   Organic_carbon   3276 non-null   float64
 7   Trihalomethanes  3114 non-null   float64
 8   Turbidity        3276 non-null   float64
 9   Potability       3276 non-null   int64  
dtypes: float64(9), int64(1)
memory usage: 256.1 KB


1) Source of data

https://www.kaggle.com/

2) Brief description of data

- The dataset shows the different measuring parameters in order to classify water quality as potable or non drinkable.

3) What is the target?

-Target is Potability

4) Is this a classification or regression problem?

-The dataset problem is close to a classification problem

5) How many features?

- Total of 9 features

6) How many rows of data.

- There are 3275 rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

- Cleaning of data and checking of missing values are some of the challenges and probably choosing the perfect model to be used.